pkg/virtctl/convert/convert.go

```diff
@@ -0,0 +1,121 @@
+package convert
+
+import (
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"fmt"
+	flag "github.com/spf13/pflag"
+)
+
+func NewConvertCommand() *Convert {
+	return &Convert{stdin: os.Stdin, stdout: os.Stdout}
+}
+
+type Convert struct {
+	stdin  io.ReadCloser
+	stdout io.Writer
+}
+
+func (c *Convert) FlagSet() *flag.FlagSet {
+	cf := flag.NewFlagSet("convert", flag.ExitOnError)
+	cf.StringP("filename", "f", "", "Filename, directory, or URL to files identifying the resource to get from a server.")
+	cf.StringP("output", "o", "json", "Output format. One of: json|yaml|xml.")
+	cf.String("output-version", "kubevirt.io/v1alpha1", "Output the formatted object with the given group version")
+	return cf
+}
+
+func (c *Convert) Usage() string {
+	usage := "Convert between Kubevirt and Libvirt VM representations:\n\n"
+	usage += "Examples:\n"
+	usage += "# Convert Domain xml to yaml via stdin\n"
+	usage += "virsh dumpxml testvm | virtctl convert-spec -f - -o yaml\n\n"
+	usage += "# Convert yaml into json from a http resource\n"
+	usage += "virtctl convert-spec -f http://127.0.0.1:4012/vm.yaml -o json\n\n"
+	usage += "# Convert VM specification from a yaml file into a Libvirt Domain XML\n"
+	usage += "virtctl convert-spec -f vm.yaml -o xml > dom.xml\n\n"
+	usage += "Options:\n"
+	usage += c.FlagSet().FlagUsages()
+	return usage
+}
+
+func (c *Convert) Run(flags *flag.FlagSet) int {
+	sourceName, _ := flags.GetString("filename")
+
+	outputFlag, _ := flags.GetString("output")
+	outputFormat := Type(outputFlag)
+
+	if sourceName == "" {
+		log.Println("No source specified")
+		return 1
```

Comment:
It might be more clear if we create enums at the top of the module along the lines of "RuntimeSuccess" and "RuntimeError"